### PR TITLE
Added contribution and github action files

### PR DESCRIPTION
**Pull request by saramonzon**
- Contributing guidelines are now described for this repo
- New github action to automatically create PRs to syncronize branches

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+## Contribution workflow
+
+If you'd like to write or modify some code for buisciii-tools, the standard workflow is as follows:
+
+1. Check that there isn't already an issue about your idea in the [buisciii-tools issues](https://github.com/BU-ISCIII/buisciii-tools/issues) to avoid duplicating work. **If there isn't one already, please create one so that others know you're working on this**
+2. [Fork](https://help.github.com/en/github/getting-started-with-github/fork-a-repo) the [buisciii-tools repository](https://github.com/BU-ISCIII/buisciii-tools/) to your GitHub account
+3. Make the necessary changes / additions within your forked repository following the [code style guidelines](#code-style-guidelines)
+4. Modify the `CHANGELOG` file according to your changes.
+5. Update any documentation as needed.
+6. [Submit a Pull Request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) against the `develop` or `hotfix` branch and sent the url to the #pipelines-dev channel in slack (if you are not in the slack channel just wait fot the PR to be reviewed and rebased).
```
Comment (Shettland) [inline .github/CONTRIBUTING.md:12]: ```against the `develop` or `hotfix` branch and sent```
sent -> send

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+## Tests
+
+When you create a pull request with changes, [GitHub Actions](https://github.com/features/actions) will run automatic tests.
+Typically, pull-requests are only fully reviewed when these tests are passing, though of course we can help out before then.
+
+There are typically two types of tests that run:
+
+### Lint tests
+
+We use black and flake8 linting based on PEP8 guidelines. You can check more information [here](https://github.com/BU-ISCIII/BU-ISCIII/wiki/Python#linting)
```
Comment (Shettland) [inline .github/CONTRIBUTING.md:40]: You could mention that this only works for python, bash scripts do not need to pass linting (don't they?)

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (saramonzon) [reply .github/CONTRIBUTING.md]: fixed thanks!

File: .github/CONTRIBUTING.md (hunk 2 above)
Comment (saramonzon) [reply .github/CONTRIBUTING.md]: Yep, no clarified in the text. Maybe there is some linting tool for bash scripting? :smiling_imp:

**Review by jaimeozaez — APPROVED** [review]

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+Anyhow you should always submit locally tested code!!
+
+### New version bumping and release
+
+In order to create a new release you need to follow the next steps:
+
+1. Set the new version according to [semantic versioning](https://semver.org/), in our particular case, changes in the `hotfix` branch will change the PATCH version (third one), and changes in develop will typicaly change the MINOR version, unless the developing team decides otherwise.
+2. Create a PR bumping the new version against `hotfix` or `develop`. For bumping a new version just change [this line](https://github.com/BU-ISCIII/buisciii-tools/blob/615f1390d96cd6c8168acebc384289520a3cd728/setup.py#L5) with the new version.
+3. Once that PR is merged, create via web another PR against `main` (origin `develop` or `hotfix` accordingly). This PR would need 2 approvals.
+4. [Create a new release](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository) copying the appropiate notes from the `CHANGELOG`.
```
Comment (jaimeozaez) [inline .github/CONTRIBUTING.md:54]: Is the link of "CHANGELOG" correct?

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
+# buisciii-tools: Contributing Guidelines
+
+## Contribution workflow
+
+If you'd like to write or modify some code for buisciii-tools, the standard workflow is as follows:
+
+1. Check that there isn't already an issue about your idea in the [buisciii-tools issues](https://github.com/BU-ISCIII/buisciii-tools/issues) to avoid duplicating work. **If there isn't one already, please create one so that others know you're working on this**
+2. [Fork](https://help.github.com/en/github/getting-started-with-github/fork-a-repo) the [buisciii-tools repository](https://github.com/BU-ISCIII/buisciii-tools/) to your GitHub account
+3. Make the necessary changes / additions within your forked repository following the [code style guidelines](#code-style-guidelines)
+4. Modify the `CHANGELOG` file according to your changes.
```
Comment (jaimeozaez) [inline .github/CONTRIBUTING.md:10]: Maybe this part would need some additional explanation.

**Review by jaimeozaez — COMMENTED** [review]
It's almost all quite clear.
I feel the CHANGELOG part would need a little bit more information (I can't even find the file).
Several lines need to end with a period (I just marked the first one I found).

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
+# buisciii-tools: Contributing Guidelines
+
+## Contribution workflow
+
+If you'd like to write or modify some code for buisciii-tools, the standard workflow is as follows:
+
+1. Check that there isn't already an issue about your idea in the [buisciii-tools issues](https://github.com/BU-ISCIII/buisciii-tools/issues) to avoid duplicating work. **If there isn't one already, please create one so that others know you're working on this**
```
[the remark this answers is not in the document]
Comment (saramonzon) [reply .github/CONTRIBUTING.md:7]: ok done! thanks!

File: .github/CONTRIBUTING.md (hunk 4 above)
Comment (saramonzon) [reply .github/CONTRIBUTING.md]: I've added some clarification for the changelog, was only that point? Or another one needs further explanation?

Comment (jaimeozaez) [reply .github/CONTRIBUTING.md]: Yes, it was just about the changelog. Thanks!

Btw, lines 7 and 8 still need a period 😇 😇

Comment (saramonzon) [reply .github/CONTRIBUTING.md]: hope i got them all already xD

**Review by svarona — APPROVED** [review]

**Pull request merged**